### ISSUE TITLE
File upload with quote in name fail

### ISSUE THREAD

**Issue by gzoritchak** [ktorio/ktor#865]
With ktor-netty 1.0.0, uploading a file with a form:

```
<form action="/file" method='post' enctype="multipart/form-data">
<input type="file"
       id="avatar" name="avatar"
       accept="image/png, image/jpeg">
<button>upload</button>
</form>
```
fail during `call.receiveMultipart`:
```
post("file"){
    val multipart = call.receiveMultipart()
    multipart.forEachPart { part ->
        if(part is PartData.FileItem) {
            val name = part.originalFileName!!
            val file = File("/uploads/$name")
            part.streamProvider().use { its ->
                file.outputStream().buffered().use {
                    its.copyTo(it)
                }
            }
        }
        part.dispose()
    }
}
```
if the filename contains a quote in the name.

```
kotlinx.io.charsets.MalformedInputException: Input length = 1
	at kotlinx.io.charsets.CharsetJVMKt.throwExceptionWrapped(CharsetJVM.kt:287)
	at kotlinx.io.charsets.CharsetJVMKt.decodeImplSlow(CharsetJVM.kt:253)
	at kotlinx.io.charsets.CharsetJVMKt.decodeExactBytes(CharsetJVM.kt:215)
	at kotlinx.io.core.StringsKt.readTextExactBytes(Strings.kt:349)
	at kotlinx.io.core.StringsKt.readTextExactBytes$default(Strings.kt:348)
	at kotlinx.io.core.ByteReadPacketBase.readText(Packet.kt:468)
	at kotlinx.io.core.ByteReadPacketBase.readText$default(Packet.kt:466)
	at io.ktor.http.cio.CIOMultipartDataBase.partToData(CIOMultipartData.kt:130)
	at io.ktor.http.cio.CIOMultipartDataBase$partToData$1.invokeSuspend(CIOMultipartData.kt)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:32)
	at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:236)
```

**Comment by gzoritchak** [comment]
As a workaround, it's possible using [javascript api ](https://developer.mozilla.org/en-US/docs/Web/API/FormData/append)to set the filename.

**Comment by cy6erGn0m** [comment]
Could you please check it with 1.1.1? At least one filename encodings related issue has been fixed after 1.0.0

**Comment by gzoritchak** [comment]
Yes, it's ok now. I should have checked. 
Thanks,
